08-regression2.Rmd

```diff
@@ -439,61 +468,263 @@ sqftc <- format(round(coeffs %>% filter(term == "sqft") %>% pull(estimate)), sci
 bedsc <- format(round(coeffs %>% filter(term == "beds") %>% pull(estimate)), scientific = FALSE)
 ```
 
-$$\text{house price} = `r icept` + `r sqftc`\cdot (\text{house size})  `r bedsc` \cdot (\text{number of bedrooms})$$
+$$\text{house sale price} = `r icept` + `r sqftc`\cdot (\text{house size})  `r bedsc` \cdot (\text{number of bedrooms})$$
 
 This model is more interpretable than the multivariate K-NN
 regression model; we can write a mathematical equation that explains how
-each predictor is affecting the predictions. But as always, we should look at
-the test error and ask whether linear regression is doing a better job of
-predicting compared to K-NN regression in this multivariate regression case. To
-do that we can use this linear regression model to predict on the test data to
-get our test error.
+each predictor is affecting the predictions. But as always, we should 
+question how well multivariate linear regression is doing compared to
+the other tools we have, such as simple linear regression 
+and multivariate K-NN regression. If this comparison is part of 
+the model tuning process&mdash;for example, if we are trying
+out many different sets of predictors for multivariate linear
+and K-NN regression&mdash;we must perform this comparison using
+cross-validation on only our training data. But if we have already
+decided on a small number (e.g., 2 or 3) of tuned candidate models and
+we want to make a final comparison, we can do so by comparing the prediction
+error of the methods on the test data.
 
 ```{r 08-get-RMSPE}
 lm_mult_test_results
 ```
 
-We get that the RMSPE for the multivariate linear regression model 
-of `r format(lm_mult_test_results %>% filter(.metric == 'rmse') %>% pull(.estimate), scientific = FALSE)`. This prediction error
+We obtain an RMSPE for the multivariate linear regression model 
```

Comment:
When we compare models in this chapter, we should compare cross-validated error, not the test error. We need to practice what we preach. This chapter would be nice to reinforce this - we don't need CV to choose a parameter, but we should still CV to compare models, not test error.

Reply:
disagree, spinning out into its own issue